src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,622 @@
+		this.stats.success = false;
+
+		const config = params.webuntisConfig;
+
+		const metaData = await this.obtainMetadata(params);
+		// await this.login(config.url, config.schoolname, config.username, config.password);
+		
+		const data = await this.fetchData(metaData);
+
+		// await this.logout();
+
+		await this.createCourses(config, data, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	getImportConditionEvaluator(params) {
+		if (params.datatype === 'inclusive') {
+			if (!params.courseMetadataIds) {
+				return ((id) => false);
+			}
+			return ((id) => params.courseMetadataIds.includes(id));
+		}
+		if (params.datatype === 'exclusive') {
+			if (!params.courseMetadataIds) {
+				return ((id) => false);
+			}
+			return ((id) => !params.courseMetadataIds.includes(id));
+		}
+		throw new BadRequest('invalid datatype');
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	async obtainMetadata(params) {
+		const metaData = await this.app.service('webuntisMetadata').find({
+			query: { datasourceId: params.datasourceId },
+			paginate: false
+		});
+
+		return metaData;
+	}
+
+	/**
+	* @return List of potential courses
+	*   Array of {
+	*     teacher: 'Renz',
+	*     class: [ '2a', '2b' ],
+	*     subject: 'mathe',
+	*     times: [ { weekDay, startTime, endTime, room } ],
+	*     state: 'new',
+	*   }
+	*/
+	async fetchMetaData() {
+		const intermediateData = {};
+		
+		intermediateData.currentSchoolYear = await this.getCurrentSchoolyear();
+
+		// To iterate over either concept
+		intermediateData.classes = await this.getClasses(intermediateData.currentSchoolYear.id);
+		intermediateData.teachers = await this.getTeachers();
+		intermediateData.rooms = await this.getRooms();
+		// intermediateData.subjects = await this.getSubjects(); // currently not required
+
+		// intermediateData.timeGrid = await this.getTimegrid(); // currently not required
+
+		intermediateData.courses = [];
+
+		// Iteration approaches currently implemented: teachers
+		if (intermediateData.teachers !== undefined) { // Iterate over teachers
+			for (const teacher of intermediateData.teachers) {
+				const currentTeacherName = `${teacher.foreName} ${teacher.longName}`;
+
+				let timetable = await this.getCustomizableTimeTableFor(2, teacher.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					roomFields: ['id', 'longname'],
+					teacherFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				let filteredTimetable = timetable.filter(entry => !(entry.te.length === 1
+					&& entry.te[0].id === teacher.id
+					&& entry.kl.length > 0
+					&& entry.ro.length === 1
+					&& entry.su.length === 1));
+				if (filteredTimetable.length > 0) {
+					filteredTimetable = filteredTimetable.map(entry => {
```

Comment:
Codacy found an issue: [Unexpected block statement surrounding arrow body; parenthesize the returned value and move it immediately after the `=>`.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-server/pullRequest?prid=6045816)